lib/router/route-info.ts

```diff
@@ -277,7 +277,7 @@ export default class InternalRouteInfo<R extends Route> {
 
     if (cached !== undefined) {
       // SAFETY: This is potentially a bit risker, but for what we're doing, it should be ok.
-      ROUTE_INFOS.set((this as unknown) as InternalRouteInfo<Route>, cached);
+      ROUTE_INFOS.set((resolved as unknown) as InternalRouteInfo<Route>, cached);
```

Comment:
This corrects a mistake I made here:
https://github.com/tildeio/router.js/commit/c83aeed0869425d2e2f412c20df9b7d6badc8d85#diff-d864d89ea3148dcf27e0d6112d2ebc5882ad17bdedaf05bfb71068d04a028438L270-R280